eng/common/pipelines/templates/steps/sparse-checkout.yml

```diff
@@ -31,6 +31,7 @@ steps:
       displayName: Init sparse checkout ${{ repo.Name }}
       workingDirectory: ${{ coalesce(repo.WorkingDirectory, format('{0}/{1}', '$(System.DefaultWorkingDirectory)', repo.Name)) }}
 
-    - pwsh: git checkout ${{ repo.Commitish }}
-      displayName: Sparse checkout at ${{ repo.Commitish }}
+    - pwsh: |
+        git checkout ${{ repo.Commitish }}  # this will use the default branch if repo.Commitish is empty
```

Comment:
How does it pick-up the default branch? Are you sure git checkout without parameters checks out the default branch? As best I can tell it will be a noop.

Reply:
From my testing (local and in-pipeline), it does. According to the [docs](https://git-scm.com/docs/git-checkout):

> You could omit <branch>, in which case the command degenerates to "check out the current branch", which is a glorified no-op with rather expensive side-effects to show only the tracking information, if exists, for the current branch.

Since we're in a state where sparse checkout paths are set but no blobs are downloaded, this seems to have the effect of grabbing all the relevant blobs.

I could switch it back to using the regex that we're using elsewhere, but as this seems to work I thought it was much cleaner.

Reply:
If this works than great. If not I was think we could use the default branch template and the $(DefaultBranch)  variable in this place.

Reply:
I actually only discovered this because I copied that DefaultBranch code over and forgot to change the remote from `Source` to `origin`, so it failed to set the variable and just ran `git checkout`, which worked!